content/blueprints/gcp.md

```diff
@@ -0,0 +1,1277 @@
+
+    *type:* string
+    *default:*
+  * `use_external_resource`
+    Indicate whether the resource exists or if Cloudify should create the resource. If set to true, this node will be static IP, otherwise ephemeral IP.
+
+    *type:* boolean
+    *default:* False
+
+
+
+
+
+## cloudify.gcp.nodes.FirewallRule
+**Derived From:** [cloudify.nodes.Root]({{< relref "blueprints/built-in-types.md" >}})
+
+A GCP FirewallRule.
+
+This describes allowed traffic directed to either the whole of the specified network, or to Instances specified by matching tags.
+
+
+
+**Properties:**
+
+
+  * `sources`
+    List of CIDR formatted ranges and instance tags which
+    will be permitted to connect to targets by this rule
+    e.g.:
+      - 10.100.101.0/24
+      - a-tag
+
+    *required* None
+  * `additional_settings`
+    Additional setting for firewall
+
+    *default:* {}
+  * `name`
+    Optional security group name. By default it will be network name plus node name.
+
+    *default:*
+  * `allowed`
+    Dictionary of allowed ports per protocol, in the form
+      protocol: [port, ...]
+    If no ports are specified then all ports are opened for that protocol eg:
+      tcp: 80, 443
+      udp:
+
+    *required* None
+  * `gcp_config`
+    A dictionary of values to pass to authenticate with the Google Cloud Platform API.
+
+    *default:* {}
+  * `target_tags`
+    List of target tags this rule should apply to. If no tags are specified, it will apply to all instances in the network
+
+    *default:* []
+  * `use_external_resource`
+    Indicate whether the resource exists or if Cloudify should create the resource.
+
+    *type:* boolean
+    *default:* False
+
+
+
+### Example
+
+{{< gsHighlight  yaml  >}}
+
+allow_ssh:
+  type: cloudify.gcp.nodes.FirewallRule
+  properties:
+    sources: [0.0.0.0/0]
+    allowed:
+      tcp: [22]
+
+allow_http_to_http_tag:
+  type: cloudify.gcp.nodes.FirewallRule
+  properties:
+    sources: [0.0.0.0/0]
+    allowed:
+      tcp: [80]
+    target_tags: [http]
+
+http_instance:
+  type: cloudify.gcp.nodes.Instance
+  properties:
+    tags: [http]
+    ...
+
+{{< /gsHighlight >}}
+
+
+
+
+
+## cloudify.gcp.nodes.GlobalAddress
+**Derived From:** [cloudify.nodes.VirtualIP]({{< relref "blueprints/built-in-types.md" >}})
+
+A GCP GlobalAddress.
+
+GlobalAddress can only be used together with GlobalForwardingRule. If you want to connect a static IP to an Instance, use StaticIP instead.
+
+
+
+**Properties:**
+
+
+  * `gcp_config`
+    A dictionary of values to pass to authenticate with the GCP API.
+
+    *default:* {}
+  * `additional_settings`
+    Additional setting for static ip
+
+    *default:* {}
+  * `use_external_resource`
+    Indicate whether the resource exists or if Cloudify should create the resource. If set to true, this node will be already existing static IP address, otherwise it will be reserved static IP address.
```

Comment:
Indicate**s** whether the resource exists **and should be used (true),** or if Cloudify should create the resource **(false).** If set to true, this node will be **an** ~~already~~ existing static IP address, otherwise it will be **a** reserved static IP address.